model/client4.go

```diff
@@ -36,10 +37,15 @@ func NewAPIv4Client(url string) *Client4 {
 }
 
 func BuildErrorResponse(r *http.Response, err *AppError) *Response {
-	if r == nil {
-		return &Response{StatusCode: 0, Error: err}
-	} else {
-		return &Response{StatusCode: r.StatusCode, Error: err}
+	statusCode := 0
+	if r != nil {
+		statusCode = r.StatusCode
+	}
+
+	return &Response{
+		StatusCode: statusCode,
+		Error:      err,
+		Response:   r,
```

Comment:
@crspeller What if I change this to pass back just the `r.Header` field then? It's just a `http.Header` object (which is a `map[string][]string`), so I could make an empty one to pass back when `r == nil`.

And @ccbrown, I don't want to remove existing fields since we're intending people to be using `client4.go` if they're writing an integration using go, and we don't want to break their code